instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -63,14 +64,30 @@ public final class OpenTelemetryDriver implements Driver {
   private static final AtomicBoolean REGISTERED = new AtomicBoolean();
   private static final List<Driver> DRIVER_CANDIDATES = new CopyOnWriteArrayList<>();
 
-  private static final SqlCommenter sqlCommenter =
-      SqlCommenter.builder()
-          .setEnabled(
-              ConfigPropertiesUtil.getBoolean(
-                  "otel.instrumentation.jdbc.experimental.sqlcommenter.enabled",
-                  ConfigPropertiesUtil.getBoolean(
-                      "otel.instrumentation.common.experimental.db-sqlcommenter.enabled", false)))
-          .build();
+  private static SqlCommenter getSqlCommenter(OpenTelemetry openTelemetry) {
+    boolean enabled =
+        DeclarativeConfigUtil.getInstrumentationConfig(openTelemetry, "jdbc")
+            .get("sqlcommenter/development")
+            .getBoolean("enabled", false);
+    if (!enabled) {
+      enabled =
+          ConfigPropertiesUtil.getBoolean(
+              "otel.instrumentation.jdbc.experimental.sqlcommenter.enabled", false);
+    }
+    if (!enabled) {
+      enabled =
+          DeclarativeConfigUtil.getInstrumentationConfig(openTelemetry, "common")
+              .get("database")
+              .get("sqlcommenter/development")
+              .getBoolean("enabled", false);
+    }
+    if (!enabled) {
+      enabled =
+          ConfigPropertiesUtil.getBoolean(
+              "otel.instrumentation.common.experimental.db-sqlcommenter.enabled", false);
+    }
```

Comment:
I think the fallback shouldn't happen if there's an explicit `false`

I think you need to use the @Nullable Boolean returning version and check for null return